backend/core/src/main/java/org/sonarsource/sonarlint/core/plugin/PluginsService.java

```diff
@@ -179,7 +179,7 @@ public boolean supportsCustomSecrets(String connectionId) {
       return false;
     }
     // when storage is not present, assume that secrets are not supported by server
-    return connection.getKind() != ConnectionKind.SONARCLOUD && storageService.connection(connectionId).serverInfo().read()
+    return connection.getKind() == ConnectionKind.SONARCLOUD || storageService.connection(connectionId).serverInfo().read()
```

Comment:
It doesn't have to be done for the release but I think a test would help. It was already missing because we did not discover the bug in the first place, now that we are fixing it, I think we should add one